src/assessments/assessments-requirements-filter.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { assessmentsProviderWithFeaturesEnabled } from 'assessments/assessments-feature-flag-filter';
+import { AutomatedChecks } from 'assessments/automated-checks/assessment';
+import { FeatureFlagStoreData } from 'common/types/store-data/feature-flag-store-data';
+
+import { AssessmentsProviderImpl } from './assessments-provider';
+import { AssessmentsProvider } from './types/assessments-provider';
+
+export function assessmentsProviderForRequirements(
+    assessmentProvider: AssessmentsProvider,
+    featureFlags: FeatureFlagStoreData,
+    requirementKeys: string[],
+): AssessmentsProvider {
+    const enabledFeaturesProvider = assessmentsProviderWithFeaturesEnabled(
```

Comment:
Nit: I feel like for now, we don't actually need this and if anything, we can assume we're getting the filtered provider in the parameters.